### PR TITLE
feat: add bigInt function with comprehensive tests

### PR DESCRIPTION
**Pull request by Daniel-Grounin**
This PR introduces the implementation of the `bigInt` function in the Number module. 
- Added support for generating random BigInt numbers within a specific range.
- Updated unit tests to cover various scenarios, including valid ranges, negative ranges, and edge cases like same min and max values.
- All tests were run and passed successfully.

Please review the changes and let me know if there are any adjustments needed. Thank you!

- Fixed line-ending issues (CRLF/LF) that were causing changes across all lines of the files, making it hard to review. Now, only the necessary changes are visible in the diff.

### CODE REVIEW

File: include/faker-cxx/number.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #include <optional>
 #include <random>
 #include <stdexcept>
+#include <optional>
```
Comment (cieslarmichal) [inline include/faker-cxx/number.h:7]: Optional is already included above

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
